crates/ark/src/lsp/symbols.rs

```diff
@@ -120,13 +120,56 @@ fn is_indexable(node: &Node) -> bool {
     true
 }
 
+// Function to parse a comment and return the section level and title
+fn parse_comment_as_section(comment: &str) -> Option<(usize, String)> {
+    // Match lines starting with one or more '#' followed by some non-empty content and must end with 4 or more '-' or '#'
+    // Ensure that there's actual content between the start and the trailing symbols.
+    let comment_re = regex::Regex::new(r"^(#+)\s*([^\s#-].+?)\s*(#{4,}|-{4,}|={4,})$").unwrap();
```

Comment:
It turns out that regex are fairly slow to compile on the fly. We typically compile them 1 time on first access and reuse them. The best example is `RE_COMMENT_SECTION`, which uses new std library features for this:
https://github.com/posit-dev/ark/blob/d6ab389bb828db56850948f10b27e3161285be47/crates/ark/src/lsp/indexer.rs#L58-L59

It turns out that `RE_COMMENT_SECTION` is trying to basically do the same thing as what you do here, so I'd just reuse that name here.